src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,210 @@
+package duke.ui;
+
+import duke.exception.DukeException;
+import duke.parser.Parser;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+import duke.tasklist.TaskList;
+
+public class Ui {
+
+    /**
+     * Prints a line 4 spaces away from the left edge of the screen to visually
+     * separate Duke's replies from user input.
+     */
+    public void printLine() {
+        System.out.printf("%64s%n", "    ____________________________________________________________");
+    }
+
+    /**
+     * Prints Duke's greeting message (bounded by lines above and below).
+     */
+    public void greet() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        printLine();
+        System.out.printf("     %s%n", "Hello! I'm Duke");
+        System.out.printf("     %s%n", "What can I do for you?");
+        printLine();
+    }
+
+    /**
+     * Prints Duke's farewell message (bounded by lines above and below).
+     */
+    public void farewell() {
```

Comment:
I think it will be hard to deduce that this method prints a farewell message. Maybe a better variable name is `printFarewellMessage`?